ACS_child_allowance_rough_draft.py

```diff
@@ -0,0 +1,178 @@
+spmu.rename(columns={"child": "spmu_children"}, inplace=True)
+spmu["new_resources"] = spmu["spm_resources"] + 1200 * spmu["spmu_children"]
+person3 = person.merge(spmu[["new_resources", "spm_id"]], on=["spm_id"])
+mdf.poverty_rate(person3, "new_resources", "spm_povthreshold", "wt")
+mdf.poverty_rate(person3, "spm_resources", "spm_povthreshold", "wt")
+just_children = person3[(person3.child == 1)]
+mdf.poverty_rate(just_children, "new_resources", "spm_povthreshold", "wt")
+mdf.poverty_rate(just_children, "spm_resources", "spm_povthreshold", "wt")
+# Columns for PUMA, child, adult, or all, deep or regular, baseline and reform
+
+
+def pov(data):
+    base = mdf.poverty_rate(data, "spm_resources", "spm_povthreshold", "wt")
+    reform = mdf.poverty_rate(data, "new_resources", "spm_povthreshold", "wt")
+    deep_base = mdf.deep_poverty_rate(data, "spm_resources", "spm_povthreshold", "wt")
+    deep_reform = mdf.deep_poverty_rate(data, "new_resources", "spm_povthreshold", "wt")
+    return pd.Series(
+        {
+            "poverty_base": base,
+            "poverty_reform": reform,
+            "deep_poverty_base": deep_base,
+            "deep_poverty_reform": deep_reform,
+        }
+    )
+
+
+grouped_puma_child = person3.groupby(["puma", "child"]).apply(pov).reset_index()
+grouped_puma = person3.groupby(["puma"]).apply(pov).reset_index()
+grouped_puma
+grouped_puma_child["pct_change"] = (
+    grouped_puma_child["poverty_base"] - grouped_puma_child["poverty_reform"]
+) / grouped_puma_child["poverty_base"]
+grouped_puma_child["pct_change_deep"] = (
+    grouped_puma_child["deep_poverty_base"] - grouped_puma_child["deep_poverty_reform"]
+) / grouped_puma_child["deep_poverty_base"]
+grouped_puma["pct_change"] = (
+    grouped_puma["poverty_base"] - grouped_puma["poverty_reform"]
+) / grouped_puma["poverty_base"]
+grouped_puma["pct_change_deep"] = (
+    grouped_puma_child["deep_poverty_base"] - grouped_puma_child["deep_poverty_reform"]
+) / grouped_puma_child["deep_poverty_base"]
+grouped_list = [grouped_puma, grouped_puma_child]
+grouped_df = pd.concat(grouped_list).reset_index()
+only_children_pumas = grouped_puma_child[(grouped_puma_child.child == 1)]
+grouped_puma_no_change = grouped_puma[(grouped_puma["pct_change"] == 0)]
+grouped_df.to_csv(r"spm_by_puma.csv")
+import plotly.express as px
+
+px.histogram(grouped_df["pct_change"])
+px.histogram(grouped_df["pct_change_deep"])
+
+# About 13% of PUMAs have no change in the poverty rate
+
+mapping_house = pd.read_csv(
+    "Mapping1.csv",
+    usecols=["Assembly District Number", "PUMA Description", "STATEFIP", "PUMA"],
+)
+mapping_senate = pd.read_csv("Mapping2.csv", usecols=["Senate District Number", "PUMA"])
+mapping_house
+mapping_senate
+mapping_merged = mapping_house.merge(
+    mapping_senate[["Senate District Number", "PUMA"]], on=["PUMA"]
+)
+mapping_merged
+### Just California
+just_california = person3[(person3.st == 6)].reset_index()
```

Comment:
```suggestion
CA_FIPS = 6
just_california = person3[(person3.st == CA_FIPS)].reset_index()
```
similarly below where using 6 (avoid hard-coding numbers and codes)